crypto/keys_test.go

```diff
@@ -70,4 +103,133 @@ func TestKeyToBytes(t *testing.T) {
 			assert.Equal(tt, keyType, kt)
 		})
 	}
+
+	for _, keyType := range GetExperimentalKeyTypes() {
+		t.Run(string(keyType)+" with pointers", func(tt *testing.T) {
+			pub, priv, err := GenerateKeyByKeyType(keyType)
+
+			assert.NoError(tt, err)
+			assert.NotEmpty(tt, pub)
+			assert.NotEmpty(tt, priv)
+
+			pubKeyBytes, err := PubKeyToBytes(&pub)
+			assert.NoError(tt, err)
+			assert.NotEmpty(tt, pubKeyBytes)
+
+			reconstructedPub, err := BytesToPubKey(pubKeyBytes, keyType)
+			assert.NoError(tt, err)
+			assert.NotEmpty(tt, reconstructedPub)
+			assert.EqualValues(tt, pub, reconstructedPub)
+
+			privKeyBytes, err := PrivKeyToBytes(&priv)
+			assert.NoError(tt, err)
+			assert.NotEmpty(tt, privKeyBytes)
+
+			reconstructedPriv, err := BytesToPrivKey(privKeyBytes, keyType)
+			assert.NoError(tt, err)
+			assert.NotEmpty(tt, reconstructedPriv)
+			assert.EqualValues(tt, priv, reconstructedPriv)
+
+			kt, err := GetKeyTypeFromPrivateKey(&priv)
+			assert.NoError(tt, err)
+			assert.Equal(tt, keyType, kt)
+		})
+	}
+}
+
+func TestDilithiumKeys(t *testing.T) {
+	t.Run("Able to generate dilithium key pairs for each mode", func(tt *testing.T) {
+		tests := []struct {
+			name    string
+			m       dilithium.Mode
+			wantErr bool
+		}{
+			{
+				"mode2",
+				dilithium.Mode2,
+				false,
+			},
+			{
+				"mode3",
+				dilithium.Mode3,
+				false,
+			},
+			{
+				"mode5",
+				dilithium.Mode5,
+				false,
+			},
+			{
+				"bad",
+				nil,
+				true,
+			},
+		}
+		for _, test := range tests {
+			tt.Run(test.name, func(ttt *testing.T) {
+				pk, sk, err := GenerateDilithiumKeyPair(test.m)
+				if test.wantErr {
+					assert.Error(ttt, err)
+					assert.Nil(ttt, pk)
+					assert.Nil(ttt, sk)
+				} else {
+					assert.NoError(ttt, err)
+					assert.NotNil(ttt, pk)
+					assert.NotNil(ttt, sk)
+				}
+			})
+		}
+	})
```

Comment:
nit: minimize branching logic inside tests. In this case I suggest separating the error case behavior into a separate test. It would make things more readable.